### PR TITLE
Add Hash derive and implement Ord and PartialOrd explicitly

### LINKED ISSUE #27: Implement the `Hash` trait for `PackageVersion`
Implement the `Hash` trait for `PackageVersion` so that the object can be used as a key for a Hash map

https://doc.rust-lang.org/std/hash/trait.Hash.html

### PR DESCRIPTION
**Pull request by John15321**
Add Hash derive and implement Ord and PartialOrd explicitly so that the `pyver`s structs can be used inside a Hash map

### CODE REVIEW

File: src/version.rs [hunk 1]
```diff
@@ -259,4 +294,71 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn test_use_package_version_as_hash_key() -> Result<()> {
```
Comment (Allstreamer) [inline src/version.rs:299]: Is this just not just putting strings into an hashmap?

Comment (John15321) [reply src/version.rs:299]: Yes, that's why it's a draft

Comment (Allstreamer) [reply src/version.rs:299]: Ah

Comment (John15321) [reply src/version.rs:299]: done

**Review by Allstreamer — APPROVED** [review]
LGTM!

**Pull request merged**